### PR TITLE
Implement unit tests for the first half of the tests

### LINKED ISSUE #4: Add unit tests for the first half of chosen files
Add unit tests for:

- [x] `capitalize.js`
- [x] `add.js`
- [x] `filter.js`
- [x] `isObject.js`
- [x] `get.js`

Unit tests should include negative and edge tests as well!

### PR DESCRIPTION
**Pull request by erictelkkala**
Closes #4

### CODE REVIEW

File: src/__tests__/filter.test.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            expect(filter(users, ({ a }) => {})).toEqual(expectedUser);
+        });
+
+        test('filters array to show empty users predicate', () => {
+            const users = [{}, {}, {}];
+
+            const expectedUser = [{}, {}, {}];
+
+            expect(filter(users, ({ a }) => {})).toEqual(expectedUser);
+        });
```
Comment (erictelkkala) [inline src/__tests__/filter.test.js:100]: What is the actual expected behavior?
Returns `[]`

File: src/__tests__/capitalize.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        test('capitalizes float 1.2 to "1.2"', () => {
+            expect(capitalize(1.2)).toBe('1.2');
+        });
+
+        test.skip('capitalizes string primitive "" to Undefined', () => {
+            expect(capitalize('')).toBe('').toHaveLength(0);
+        });
+
+        // SKIPPED DUE TO BUGS
```
Comment (joniriikola) [inline src/__tests__/capitalize.test.js:40]: is this also skipped due to bug?

File: src/__tests__/filter.test.js (hunk 1 above)
Comment (joniriikola) [reply src/__tests__/filter.test.js:100]: Based on documentation seems like this is intended behavior.

File: src/__tests__/filter.test.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            const users = [
+                { user: 'barney', active: true, wallet: { money: 0 } },
+                { user: 'fred', active: false, wallet: { money: 10 } },
+            ];
+
+            const expectedUser = [];
+
+            expect(filter(users, () => {})).toEqual(expectedUser);
+        });
+    });
```
Comment (joniriikola) [inline src/__tests__/filter.test.js:68]: Specify why these tests are skipped.

File: src/__tests__/isObject.test.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    describe('negative tests', () => {
+        test('null is not an object', () => {
+            expect(isObject(null)).toBe(false);
+        })
+
+        test('undefined is not an object', () => {
+            expect(isObject(undefined)).toBe(false);
+        })
+
+        test('Primitive String is an object', () => {
```
Comment (joniriikola) [inline src/__tests__/isObject.test.js:48]:
```suggestion
        test('Primitive String is not an object', () => {
```

**Review by joniriikola — CHANGES_REQUESTED** [review]
Some minor changes and points

File: src/__tests__/filter.test.js (hunk 1 above)
Comment (erictelkkala) [reply src/__tests__/filter.test.js:100]: I'll change the test to expect just an empty array

Comment (erictelkkala) [reply src/__tests__/filter.test.js]: Fixed in commit 6bd857d86660357152bf966b6e5f204e38bbb3cb

File: src/__tests__/capitalize.test.js (hunk 2 above)
Comment (erictelkkala) [reply src/__tests__/capitalize.test.js]: Fixed in commit 4dc3481e7900e61b77d7a475529afcc449103abb

File: src/__tests__/filter.test.js (hunk 3 above)
Comment (erictelkkala) [reply src/__tests__/filter.test.js]: Fixed in commit f16f52e51ea93cf778d5cbf7769aed2312b951e3

File: src/__tests__/isObject.test.js (hunk 4 above)
Comment (erictelkkala) [reply src/__tests__/isObject.test.js]: Fixed in commit 1faa96e4

**Review by joniriikola — APPROVED** [review]

**Pull request merged**